manifests/params.pp

```diff
@@ -22,7 +22,10 @@
       $virtinst_package = 'virtinst'
       $radvd_package = 'radvd'
       $sysconfig = false
-      $deb_default = {}
+      $deb_default = $::service_provider ? {
+        'systemd' => { 'libvirtd_opts' => '' },  # no '-d', it confuses systemd
```

Comment:
Are you sure that this is sufficient? on the 16.04 box I have here service_provider is 'debian' not 'systemd'. Could be a version thing, this is with Puppet 3.8.5, Facter 2.4.6, Stdlib 4.12, masterless.